<launch>
	<arg name="boxes_topic"					default="/darknet_ros/bounding_boxes"/>
	<arg name="srcimg_topic"				default="/pgr/image_color"/>
	<!-- <arg name="srcimg_topic"				default="/ardrone/front/image_rect_color"/> -->
	<arg name="threat_image_topic"		default="/threats/potential_images"/>
	<arg name="skeleton_image_topic"	default="/threats/skeleton_images"/>

	<include file="$(find stop)/launch/stop.launch">
		<arg name="launcher"	value="evaluator_stopper" />
	</include>

	<include file="$(find usma_threat_ros)/launch/pgr.launch"/>

  <!-- Start the node to run darknet ros -->
  <include file="$(find usma_threat_ros)/launch/alexey_darknet.launch"/>

	<!-- Start the node to evaluate bounding boxes   -->
	<node pkg="usma_threat_ros" type="evaluator.py" name="threat_evaluator" output="screen" >
		<param name="yolo_boxes_topic" 	value="$(arg boxes_topic)" />
		<param name="srcimg_topic"			value="$(arg srcimg_topic)" />
		<param name="threat_images"			value="$(arg threat_image_topic)" />
	</node>

	<!-- Start the node to calculate skeletons   -->
	<include file="$(find usma_threat_ros)/launch/openpose.launch">
		<arg name="threat_image_topic" 		value="$(arg threat_image_topic)" />
		<arg name="skeleton_image_topic" 	value="$(arg skeleton_image_topic)" />
	</include>

	<node pkg="image_view" type="image_view" name="detection_image" >
		<remap from="image" to="/darknet_ros/detection_image"/>
	</node>

	<node pkg="image_view" type="image_view" name="skeleton_images" >
		<remap from="image" to="/threats/skeleton_images"/>
	</node>
</launch>
